AstSemantics.md

```diff
@@ -294,36 +294,40 @@ buffer.
 In the MVP, the length of the return types sequence may only be 0 or 1. This
 restriction may be lifted in the future.
 
-Direct calls to a function specify the callee by index into a function table.
+Direct calls to a function specify the callee by index into a *main function table*.
 
   * `call`: call function directly
 
-Calls must match the function signature exactly.
+A direct call to a function with a mismatched signature is a module verification error.
 
 Like direct calls, calls to [imports](Modules.md#imports-and-exports) specify
-the callee by index into an import table (defined by the sequence of import
-declarations in the module import section) and the call must match the declared
-signature of the import exactly.
+the callee by index into an *imported function table* defined by the sequence of import
+declarations in the module import section. A direct call to an imported function with a
+mismatched signature is a module verification error.
 
   * `call_import` : call imported function directly
 
-Indirect calls may be made to a value of function-pointer type. A 
-function-pointer value may be obtained for a given function as specified by its index
-in the function table.
+Indirect calls allow calling target functions that are unknown at compile time.
+The target function is an expression of local type `i32` and is always the first
```

Comment:
Should this be `i64` for wasm64? I don't have a specific use case that needs more than 4 billon indirect function table entries, but since functions can appear multiple times in the table, one could imagine possibilities.